src/components/PopupConfirm.js

```diff
@@ -0,0 +1,22 @@
+import Popup from "./Popup.js";
+
+export default class PopupConfirm extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._imageEditForm = this._popupElement.querySelector(".modal__form");
+  }
+
+  setSubmitAction(handler) {
+    this._handleSubmit = handler;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+
+    this._imageEditForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      this._handleSubmit();
+      this.close();
```

Comment:
Needs correction: Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form again.